core/cat/rabbit_hole.py

```diff
@@ -217,36 +217,19 @@ def file_to_docs(
         blob = Blob(data=file_bytes,
                     mimetype=content_type,
                     source=source).from_data(data=file_bytes,
-                                             mime_type=content_type)
+                                             mime_type=content_type,
+                                             path=source)
```

Comment:
Please don't kill me Piero, I also added the path here because it doesn't arrive any source to the parsers. As the [documentation](https://api.python.langchain.com/en/latest/document_loaders/langchain.document_loaders.blob_loaders.schema.Blob.html#langchain.document_loaders.blob_loaders.schema.Blob.from_data) says, `path` will set the source of the Blob